### PR TITLE
[9.x] Drop CommonMark v1 support

### PR DESCRIPTION
**Pull request by driesvints**
~The changes are straightforward but because of namespace changes to the `Environment` class we'll need to target v9. I think I can get CommonMark v2 working on 8.x but it'll require some nasty hacks. Feel free to let me know if you want to check into that.~

Drops CommonMark v1 support for Laravel v9

### CODE REVIEW

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -401,9 +404,13 @@ public static function words($value, $words = 100, $end = '...')
...
     public static function markdown($string, array $options = [])
     {
-        $converter = new GithubFlavoredMarkdownConverter($options);
+        $environment = new Environment($options);
+        $environment->addExtension(new CommonMarkCoreExtension());
+        $environment->addExtension(new GithubFlavoredMarkdownExtension());
 
-        return $converter->convertToHtml($string);
+        $converter = new MarkdownConverter($environment);
+
```
Comment (colinodell) [inline src/Illuminate/Support/Str.php:412]: You could revert these changes since the `GithubFlavoredMarkdownConverter` will continue to work as-is, and you're not adding any additional extensions here.  But there's no reason you can't take this more verbose route either :)

File: src/Illuminate/Mail/Markdown.php [hunk 2]
```diff
@@ -104,15 +105,16 @@ public function renderText($view, array $data = [])
...
+        $environment->addExtension(new CommonMarkCoreExtension);
         $environment->addExtension(new TableExtension);
 
-        $converter = new CommonMarkConverter([
-            'allow_unsafe_links' => false,
-        ], $environment);
+        $converter = new MarkdownConverter($environment);
 
-        return new HtmlString($converter->convertToHtml($text));
+        return new HtmlString($converter->convertToHtml($text)->getContent());
```
Comment (colinodell) [inline src/Illuminate/Mail/Markdown.php:117]: LGTM!

**Review by colinodell — APPROVED** [review]
Overall this looks perfect to me!

And I do have some uncommitted changes where both 1.x and 2.x work on Laravel 8.x without any nasty hacks if you'd like me to push that up :)

**Comment by driesvints** [comment]
Thanks @colinodell! I'll close mine in favor of yours 👍

**Comment by driesvints** [comment]
Just realized we can indeed rebase this one once yours has been merged.

File: src/Illuminate/Support/Str.php (hunk 1 above)
Comment (driesvints) [reply src/Illuminate/Support/Str.php]: Done.

**Review by colinodell — APPROVED** [review]
LGTM!

File: composer.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
         "doctrine/inflector": "^2.0",
         "dragonmantank/cron-expression": "^3.1",
         "egulias/email-validator": "^3.1",
-        "league/commonmark": "^1.3|^2.0",
```
Comment (GrahamCampbell) [inline composer.json:25]: the other composer.json files need to be updated too

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: composer.json (hunk 3 above)
Comment (driesvints) [reply composer.json:25]: Thanks, forgot about that.

**Review by GrahamCampbell — APPROVED** [review]
